### PR TITLE
[Pcre2] Check if files exists before call vcpkg_replace_string.

### PR DESCRIPTION
**Pull request by larshg**
**Describe the pull request**

- #### What does your PR fix?  
  Fixes build error if only building release.
  See here for reference: https://dev.azure.com/PointCloudLibrary/pcl/_build/results?buildId=20525&view=logs&j=645d9b2d-e6a8-55a6-fe5b-40842942afe7&t=07185e79-ad4d-5c13-c1d1-201b01637442

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Is this necessary for portfile.cmake changes?

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by dg0yt** [comment]
If this is meant to fix single-config builds, I would prefer to
- just guard the debug variant (debug-only is not possible),
- not use `if(EXISTS ...)`, but `if(NOT DEFINED VCPKG_BUILD_TYPE)` (the controlling variable).

**Comment by larshg** [comment]
> If this is meant to fix single-config builds, I would prefer to
> 
>     * just guard the debug variant (debug-only is not possible),

Why is this not possible?

> 
>     * not use `if(EXISTS ...)`, but `if(NOT DEFINED VCPKG_BUILD_TYPE)` (the controlling variable).

Sure, I'll change it.

**Comment by dg0yt** [comment]
> Why is this not possible?

By design. The most recent discussions ended here:
https://github.com/microsoft/vcpkg/pull/21206#issuecomment-969123466

File: ports/pcre2/portfile.cmake [hunk 1]
```diff
@@ -55,8 +55,10 @@ file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
 if(BUILD_STATIC)
     file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin" "${CURRENT_PACKAGES_DIR}/debug/bin")
 elseif(VCPKG_TARGET_IS_WINDOWS)
-    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/bin/pcre2-config" "${CURRENT_PACKAGES_DIR}" "`dirname $0`/..")
-    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/debug/bin/pcre2-config" "${CURRENT_PACKAGES_DIR}" "`dirname $0`/../..")
+	vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/bin/pcre2-config" "${CURRENT_PACKAGES_DIR}" "`dirname $0`/..")
+	if(NOT DEFINED VCPKG_BUILD_TYPE)
+		vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/debug/bin/pcre2-config" "${CURRENT_PACKAGES_DIR}" "`dirname $0`/../..")
+	endif()
```
Comment (JackBoosY) [inline ports/pcre2/portfile.cmake:61]:
```suggestion
    vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/bin/pcre2-config" "${CURRENT_PACKAGES_DIR}" "`dirname $0`/..")
    if(EXISTS "${CURRENT_PACKAGES_DIR}/debug/bin/pcre2-config)
        vcpkg_replace_string("${CURRENT_PACKAGES_DIR}/debug/bin/pcre2-config" "${CURRENT_PACKAGES_DIR}" "`dirname $0`/../..")
    endif()
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/pcre2/portfile.cmake (hunk 1 above)
Comment (JackBoosY) [reply ports/pcre2/portfile.cmake:61]: Use `EXISTS` is more safety here.

**Review by JonLiu1993 — APPROVED** [review]

**Review by BillyONeal — APPROVED** [review]

**Review by JonLiu1993 — APPROVED** [review]

**Review by JackBoosY — APPROVED** [review]

**Pull request merged**